text/0000-near-shell.md

```diff
@@ -0,0 +1,452 @@
+[shell-experience-settings]: #shell-experience-settings
+
+---
+
+These are key-values reflecting how `near-shell` behaves when called within a NEAR project.
+    
+Example: a developer using OS X uses `near login` with access to a browser, whereas a validator runs `near login` on a CentOS box with no UI or browser. A user prompt may ask "Is this login from a computer with a browser?" The answer is then stored in key-value format in the project-level directory so it can be referenced later, skipping the prompt in the future.
+
+Besides answers to user prompts, shell-experience settings also store information about the last version of `near-shell` used in this project. For more information, please see [Upgradability](#upgradability).
+
+As shown in the directory structure above, this file is located in the project directory at:
+
+`.near-config/settings.js`
+
+In summary, "settings" are key-value pairs that are set by `near-shell` and relate to the behavior of how it operates in a given project directory. They do not relate to the development or deployment of smart contracts, except for providing default values like `accountId`.
+
+**Note**: shell-experience settings can exist in the user's home directory as well. In this case, projects without their own `.near-config/settings.js` will use the home directory settings. `near-shell` will be quite verbose when running a command, letting the user know which settings and configuration files are being used when the user runs commands.
+
+To make this more clear, there are two nested arrays that differentiate settings that should only be applied from a project-level file from those in the home directory. This is how a `settings.js` file may look:
+
+```javascript
+{
+  …
+  "alwaysRunMigrations": true,
+  "project-only": {
+    "lastShellVersion": "0.19.1"
+  }
+}
+```
+
+As defined later, there is a command `near settings set-default` which copies the current project's settings into the home directory. Keys inside the `project-only` object will not be moved.
+
+If a new project is created and has no project-level settings file yet, and the home directory has a settings file, the home directory's file will be used as the template. Otherwise, a project-level settings file will be created and populated based on the user prompts given during regular usage of `near-shell`. 
+    
+#### 2. Configuration
+
+---
+
+Configuration are key-value pairs that relate to development and deployment of smart contracts and refer to *connection* information.
+
+Example: As a user, I want to deploy a contract to my localnet instead of testnet during development. I will provide flags and/or modify configuration so that `near-shell` can determine where to connect, for what contract, and on behalf of which NEAR account.
+
+Contains:
+
+* `networkId`       : Similar to an environment to work on (ex: 'staging')
+* `nodeUrl`         : URL to the NEAR node
+* `contractName`    : NEAR account name for the smart contract
```

Comment:
I doubt this should be here. It's commonly included in `src/config.js` today, but that doesn't seem right to me. 

Let's consider your example command below for calling a contract:

    near contract view near-game topPlayers --env localnet

What would having `contractName` in my config file give me? It doesn't make sense to call a function on a contract without specifying which contract. I doubt any sort of "default contract" makes sense.